File: Excel/500-599/583/583 Challenge.py
import pandas as pd

path = "583 Max for Cities & Birds Combination.xlsx"
input = pd.read_excel(path, usecols="A:F", skiprows=1, nrows=8)
test = pd.read_excel(path, usecols="H:I", skiprows=1, nrows=5)

input_long = input.melt(id_vars=input.columns[0], var_name="City", value_name="Count")
result = input_long[input_long['Count'] == input_long.groupby('City')['Count'].transform('max')].groupby('City').agg(
    Max=('Birds/City', ', '.join)
).reset_index().sort_values('City')

print(result.equals(test)) # True

File: Excel/500-599/583/583 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/583 Max for Cities & Birds Combination.xlsx"
input = read_excel(path, range = "A2:F10")
test  = read_excel(path, range = "H2:I7")

result = input %>%
  pivot_longer(cols = -c(1), names_to = "City", values_to = "Count") %>%
  mutate(max_per_city = max(Count), .by = City) %>%
  filter(Count == max_per_city) %>%
  summarise(Max = paste(`Birds/City`, collapse = ", "), .by  = City) %>%
  arrange(City)

all.equal(result, test)
#> [1] TRUE